source/extensions/filters/network/dubbo_proxy/router/route_matcher.cc

```diff
@@ -0,0 +1,218 @@
+    const envoy::config::filter::network::dubbo_proxy::v2alpha1::Route& route)
+    : cluster_name_(route.route().cluster()) {
+  for (const auto& header_map : route.match().headers()) {
+    config_headers_.emplace_back(header_map);
+  }
+
+  if (route.route().cluster_specifier_case() ==
+      envoy::config::filter::network::dubbo_proxy::v2alpha1::RouteAction::kWeightedClusters) {
+    total_cluster_weight_ = 0UL;
+    for (const auto& cluster : route.route().weighted_clusters().clusters()) {
+      weighted_clusters_.emplace_back(std::make_shared<WeightedClusterEntry>(*this, cluster));
+      total_cluster_weight_ += weighted_clusters_.back()->clusterWeight();
+    }
+    ENVOY_LOG(debug, "dubbo route matcher: weighted_clusters_size {}", weighted_clusters_.size());
+  }
+}
+
+const std::string& RouteEntryImplBase::clusterName() const { return cluster_name_; }
+
+const RouteEntry* RouteEntryImplBase::routeEntry() const { return this; }
+
+RouteConstSharedPtr RouteEntryImplBase::clusterEntry(uint64_t random_value) const {
+  if (weighted_clusters_.empty()) {
+    ENVOY_LOG(debug, "dubbo route matcher: weighted_clusters_size {}", weighted_clusters_.size());
+    return shared_from_this();
+  }
+
+  return WeightedClusterUtil::pickCluster(weighted_clusters_, total_cluster_weight_, random_value,
+                                          false);
+}
+
+bool RouteEntryImplBase::headersMatch(const Http::HeaderMap& headers) const {
+  ENVOY_LOG(debug, "dubbo route matcher: headers size {}, metadata headers size {}",
+            config_headers_.size(), headers.size());
+  return Http::HeaderUtility::matchHeaders(headers, config_headers_);
+}
+
+RouteEntryImplBase::WeightedClusterEntry::WeightedClusterEntry(const RouteEntryImplBase& parent,
+                                                               const WeightedCluster& cluster)
+    : parent_(parent), cluster_name_(cluster.name()),
+      cluster_weight_(PROTOBUF_GET_WRAPPED_REQUIRED(cluster, weight)) {}
+
+ParameterRouteEntryImpl::ParameterRouteEntryImpl(
+    const envoy::config::filter::network::dubbo_proxy::v2alpha1::Route& route)
+    : RouteEntryImplBase(route) {
+  for (auto& config : route.match().method().params_match()) {
+    parameter_data_list_.emplace_back(config.first, config.second);
+  }
+}
+
+ParameterRouteEntryImpl::~ParameterRouteEntryImpl() {}
+
+bool ParameterRouteEntryImpl::matchParameter(const std::string& request_data,
+                                             const ParameterData& config_data) const {
+  switch (config_data.match_type_) {
+  case Http::HeaderUtility::HeaderMatchType::Value:
+    return config_data.value_.empty() || request_data == config_data.value_;
+  case Http::HeaderUtility::HeaderMatchType::Range: {
+    int64_t value = 0;
+    return StringUtil::atol(request_data.c_str(), value, 10) &&
+           value >= config_data.range_.start() && value < config_data.range_.end();
+  }
+  default:
+    NOT_REACHED_GCOVR_EXCL_LINE;
```

Comment:
You don't have validate rule so I don't think this is not reachable? A bad config may hit this.

Reply:
I added validation for the RouteConfiguration rule.